Mailer/app/api/v1/Handlers/Mailers/MailTypesListingHandler.php

```diff
@@ -18,19 +20,36 @@ public function __construct(
         $this->listsRepository = $listsRepository;
     }
 
-    public function handle($params)
+    public function handle($payload)
     {
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'mail-types.schema.json');
+
+        if ($this->hasErrorResponse()){
+            return $this->getErrorResponse();
+        }
+
         $results = $this->listsRepository->all();
 
-        $output = [];
-        foreach ($results as $row) {
-            $item = new \stdClass();
-            $item->id = $row->id;
-            $item->code = $row->code;
-            $item->title = $row->title;
-            $output[] = $item;
+        if (isset($payload['public_listing'])){
+            $results->where(['public_listing' => $payload['public_listing']]);
         }
 
-        return new JsonApiResponse(200, ['status' => 'ok', 'data' => $output]);
+        $output = Arrays::map($results->fetchAll(), function ($row){
+            return [
```

Comment:
Are all these fields actually required in CRM? If we put them here and we realize they're not necessary / we don't want them there, it would be a breaking change to remove them.

Reply:
Shouldn't be the API also for other uses besides CRM?

Reply:
It should. But unless it's required by someone, I'd rather add an extra field (new feature) opposing to removing a field (breaking change for all implementations).

Reply:
But they are in fact all needed except for the _code_ which is in my opinion useful so I'll leave it there.

Reply:
OK then. Thanks.